devito/mpi/halo_scheme.py

```diff
@@ -113,15 +117,27 @@ def union(self, halo_schemes):
             for k, v in i.fmapper.items():
                 hse = fmapper.setdefault(k, v)
 
-                # The `loc_indices` must match
                 if hse.loc_indices != v.loc_indices:
-                    raise ValueError("Cannot compute the union of one or more HaloScheme "
-                                     "when the `loc_indices` differ")
+                    # The `loc_dirs` must match otherwise it'd be a symptom there's
+                    # something horrinbly broken somewhere!
```

Comment:
typos

Reply:
yes, fixing (these and those from the old PR in profiling.py, as promised)